model/event/subscriptions.go

```diff
@@ -38,28 +36,73 @@ type Subscription struct {
 	Selectors      []Selector    `bson:"selectors,omitempty"`
 	RegexSelectors []Selector    `bson:"regex_selectors,omitempty"`
 	Subscriber     Subscriber    `bson:"subscriber"`
+	ExtraData      interface{}   `bson:"extra_data,omitempty"`
+}
+
+type unmarshalSubscription struct {
+	ID             bson.ObjectId `bson:"_id"`
+	Type           string        `bson:"type"`
+	Trigger        string        `bson:"trigger"`
+	Selectors      []Selector    `bson:"selectors,omitempty"`
+	RegexSelectors []Selector    `bson:"regex_selectors,omitempty"`
+	Subscriber     Subscriber    `bson:"subscriber"`
+	ExtraData      bson.Raw      `bson:"extra_data,omitempty"`
+}
+
+func (s *Subscription) SetBSON(raw bson.Raw) error {
+	temp := unmarshalSubscription{}
+
+	if err := raw.Unmarshal(&temp); err != nil {
+		return errors.Wrap(err, "error unmarshalling subscriber")
+	}
+
+	if data := registry.GetExtraData(temp.Type, temp.Trigger); data != nil {
+		s.ExtraData = data
+	}
+
+	// if ExtraData is a bson null
+	if temp.ExtraData.Kind == 0x0A {
+		if s.ExtraData != nil {
+			s.ExtraData = nil
+			return errors.New("error unmarshaling extra data: expected extra data in subscription; found none")
+		}
+
+	} else {
+		if s.ExtraData == nil {
```

Comment:
Will we ever hit this case? https://github.com/evergreen-ci/evergreen/pull/1085/files#diff-f62bbd392c5d1a3e3f317c25fd2c81b5R59 checks that this is not nil.

Reply:
This would be hit in the case where the database has some extra data saved for the subscription, but Evergreen itself is not aware of the extra data